src/coordinator/roles.spec.js

```diff
@@ -29,6 +29,200 @@ describe('Coordinator/Roles', function() {
+      it('throws an error', function() {
+        const organization = fixture.build('contxtOrganization');
+        const application = fixture.build('contxtApplication');
+
+        const roles = new Roles(baseSdk, baseRequest, expectedHost);
+        const promise = roles.addApplication(organization.id, null, application.id);
+
+        return expect(promise).to.be.rejectedWith(
+          'A roleId is required for adding an application to a role'
+        );
+      });
+    });
+
+    context('when the application ID is not provided', function() {
+      it('throws an error', function() {
+        const organization = fixture.build('contxtOrganization');
+        const role = fixture.build('contxtRole');
+
+        const roles = new Roles(baseSdk, baseRequest, expectedHost);
+        const promise = roles.addApplication(organization.id, role.id, null);
+
+        return expect(promise).to.be.rejectedWith(
+          'An applicationId is required for adding an application to a role'
+        );
+      });
+    });
+  });
+
+  describe('addStack', function() {
+    context(
+      'when all the required parameters are provided and valid',
+      function() {
+        let expectedRoleStack;
+        let promise;
+        let request;
+        let stack;
+        let organization;
+        let role;
+        let roleStackFromServer;
+        let toCamelCase;
+
+        beforeEach(function() {
+          organization = fixture.build('contxtOrganization');
+          stack = fixture.build('contxtStack');
+          role = fixture.build('contxtRole');
+
+          expectedRoleStack = fixture.build('contxtRoleStack');
+          roleStackFromServer = fixture.build(
+            'contxtRoleStack',
+            expectedRoleStack,
+            {
+              fromServer: true
+            }
+          );
+
+          request = {
+            ...baseRequest,
+            post: this.sandbox.stub().resolves(roleStackFromServer)
+          };
+          toCamelCase = this.sandbox
+            .stub(objectUtils, 'toCamelCase')
+            .callsFake((app) => expectedRoleStack);
+
+          const roles = new Roles(baseSdk, request, expectedHost);
+          promise = roles.addStack(
+            organization.id,
+            role.id,
+            stack.id,
+            expectedRoleStack.accessType
+          );
+        });
+
+        it('posts the role stack to the server', function() {
+          expect(request.post).to.be.calledWith(
+            `${expectedHost}/stacks/${stack.id}/roles/${role.id}`,
+            {
+              access_type: expectedRoleStack.accessType
+            }
+          );
+        });
+
+        it('formats the returned role stack', function() {
+          return promise.then(() => {
+            expect(toCamelCase).to.be.calledWith(roleStackFromServer);
+          });
+        });
+
+        it('returns a fulfilled promise', function() {
+          return expect(promise).to.be.fulfilled;
+        });
```

Comment:
Probably needs a test about it actually returning the role stack?

Reply:
done